File: cases/taint/java/loop-carried-positive/LoopCarriedPositive.java
package dataflowbench.taint;

final class LoopCarriedPositive {
    static int dfb_source() { // DFB-SOURCE: loop-carried-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: loop-carried-sink

    static void run() {
        int value = dfb_source();
        for (int iteration = 0; iteration < 3; iteration++) {
            value = value + iteration; // DFB-WITNESS: loop-carried-value
        }
        dfb_sink(value);
    }
}

File: cases/taint/kotlin/loop-carried-positive/LoopCarriedPositive.kt
package dataflowbench

object LoopCarriedPositive {
    fun dfb_source(): Int { // DFB-SOURCE: loop-carried-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: loop-carried-sink

    fun run() {
        var value = dfb_source()
        for (iteration in 0 until 3) {
            value = value + iteration // DFB-WITNESS: loop-carried-value
        }
        dfb_sink(value)
    }
}
